jax/experimental/mosaic/gpu/profiler.py

```diff
@@ -109,6 +98,90 @@ def run(*args, **kwargs):
   return outs, float(elapsed)
 
 
+def _measure_cupti(f, aggregate):
+  def wrapper(*args, **kwargs):
+    mosaic_gpu_lib._mosaic_gpu_ext._cupti_init()
+    try:
+      results = jax.block_until_ready(jax.jit(f)(*args, **kwargs))
+    finally:
+      timings = mosaic_gpu_lib._mosaic_gpu_ext._cupti_get_timings()
+    if not timings:
+      return results, None
+    elif aggregate:
```

Comment:
Is it useful to have `aggregate=False`? Can we assume its true so that the return type of `measure` is the same for both modes?

Reply:
Definitely useful, I'd rather keep it.

Reply:
The user can always aggregate manually, it's just one comprehension away.

Curious what @apaszke thinks as well.

Reply:
> The user can always aggregate manually

This seems to suggest "let's make `aggregate=False` the only option" because the user can aggregate themselves, but then the return types are going to be different between the two modes.

> Can we assume its true so that the return type of measure is the same for both modes?

This seems to suggest making `aggregate=True` the only option to make the return types uniform.

Am I missing something or these are contradictory?

I think it's valuable to be able to look at precise individual (disaggregated) kernel timings, this is a crucial bit of functionality that you can only get with CUPTI and not with events. Keeping it is more important than making the return types uniform.

We settled on this design (aggregate/summed timings by default with an option to see individual timings) with @apaszke over DMs over a month ago, but I should have posted more widely so we could have had this discussion earlier :)